include/bm/bm_sim/actions.h

```diff
@@ -545,6 +552,23 @@ ActionParam::to<const std::vector<Data>>(ActionEngineState *state) const {
   return vec;
 }
 
+template <> inline
+const std::vector<Field>
+ActionParam::to<const std::vector<Field>>(ActionEngineState *state) const {
+  _BM_ASSERT(tag == ActionParam::FIELD_LIST && "not a field list");
+  std::vector<Field> vec;
```

Comment:
I believe that we don't actually need to make copy of the fields, we could store "references" in the vector but since actually storing references in a C++ vector is not allowed as they are not assignable, this is not a trivial change. It can be addressed in a future PR.